credentials/tls/certprovider/pemfile/watcher.go

```diff
@@ -191,23 +198,58 @@ func (w *watcher) updateRootDistributor() {
 		return
 	}
 
+	// If SPIFFEBundleMap is set, use it and DON'T use the RootFile, even if it
+	// fails
+	if w.opts.SPIFFEBundleMapFile != "" {
+		w.maybeUpdateSPIFFEBundleMap()
+	} else {
+		w.maybeUpdateRootFile()
+	}
+}
+
+func (w *watcher) maybeUpdateSPIFFEBundleMap() error {
+	// If the map file is unset, just return an error, don't create log spam.
+	if w.opts.SPIFFEBundleMapFile == "" {
+		return fmt.Errorf("SPIFFEBundleMapFile is unset in watcher options")
+	}
+	spiffeBundleMapContents, err := os.ReadFile(w.opts.SPIFFEBundleMapFile)
+	if err != nil {
+		logger.Warningf("spiffeBundleMapFile (%s) read failed: %v", w.opts.SPIFFEBundleMapFile, err)
+		return err
+	}
+	// If the file contents have not changed, skip updating the distributor.
+	if bytes.Equal(w.spiffeBundleMapFileContents, spiffeBundleMapContents) {
+		return nil
+	}
+	bundleMap, err := spiffe.BundleMapFromBytes(spiffeBundleMapContents)
+	if err != nil {
+		logger.Warning("failed to parse spiffe bundle map")
```

Comment:
Nit: s/failed/Failed/

I see that there are existing violations of this style guide recommendation in this file, but at least lets not add more. See: go/go-style/decisions#error-strings